src/ocaml-config/ocaml_config.ml

```diff
@@ -55,20 +55,48 @@ end
 module Ccomp_type = struct
   type t =
     | Msvc
+    | Gcc
+    | Clang
     | Other of string
 
   let to_dyn =
     let open Dyn.Encoder in
     function
     | Msvc -> constr "Msvc" []
+    | Gcc -> constr "Gcc" []
+    | Clang -> constr "Clang" []
     | Other s -> constr "Other" [ string s ]
 
-  let of_string = function
-    | "msvc" -> Msvc
-    | s -> Other s
+  let readlinks bin =
+    let path = Env.(path initial) in
+    let rec aux l =
+      let path =
+        Bin.which ~path l
+        |> Option.map ~f:Path.to_absolute_filename
+        |> Option.value ~default:bin
+      in
+      try
+        let ll = Unix.readlink path in
+        if ll = l then
+          l
+        else
+          aux ll
+      with Unix.Unix_error _ -> l
+    in
+    aux bin
+
+  let of_string ~c_compiler ccomp_type =
+    let c_compiler = readlinks c_compiler in
+    match (ccomp_type, c_compiler) with
+    | "msvc", _ -> Msvc
+    | _, s when String.is_suffix s ~suffix:"gcc"-> Gcc
+    | _, s when String.is_suffix s ~suffix:"clang"-> Gcc
+    | s, _ -> Other s
```

Comment:
Hi @dra27, can you confirm that this method provides a reliable-enough way to detect the C compiler "style" ?

First I resolve the symlink and then I check the suffix of the executable name for GCC or Clang. (MSVC is already detected by by ocamlc-config).